SPECS/symcrypt/symcrypt.spec

```diff
@@ -0,0 +1,56 @@
+%define debug_package %{nil}
+Summary:        A core cryptographic library written by Microsoft
+Name:           symcrypt
+Version:        100.20
+Release:        1%{?dist}
+License:        MIT License
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          System/Libraries
+URL:            https://github.com/microsoft/SymCrypt
+Source0:        %{name}-%{version}-with-submodules.tar.gz
+Patch0:         build-only-generic.patch
+Patch1:         tests-build-only-generic.patch
+BuildRequires:  cmake
+BuildRequires:  gcc
+BuildRequires:  make
+BuildRequires:  python3
+BuildRequires:  python3-pyelftools
+
+%description
+A core cryptographic library written by Microsoft
+
+%prep
+%setup -n SymCrypt
+
+%patch0
+%patch1
+
+%build
+mkdir bin
+cd bin
+%ifarch aarch64
+cmake .. -DCMAKE_TOOLCHAIN_FILE="../cmake-toolchain/LinuxUserMode-ARM64.cmake" -DCMAKE_BUILD_TYPE=Release
+%else
+cmake .. -DCMAKE_TOOLCHAIN_FILE="../cmake-toolchain/LinuxUserMode-AMD64.cmake" -DCMAKE_BUILD_TYPE=Release
+%endif
+cmake --build .
+
+%install
+mkdir -p %{buildroot}%{_libdir}
+mkdir -p %{buildroot}%{_includedir}
+%ifarch aarch64
+install bin/module/ARM64/LinuxUserMode/generic_linux/libsymcrypt.so %{buildroot}%{_libdir}/
+%else
+install bin/module/AMD64/LinuxUserMode/generic_linux/libsymcrypt.so %{buildroot}%{_libdir}/
+%endif
+install inc/* %{buildroot}%{_includedir}
+
+%files
+%license LICENSE
+%{_libdir}/libsymcrypt.so
```

Comment:
Is this name correct? Commonly in Linux libraries are versioned, plain `A.so` files are just links and actual binaries have a form similar to `A.so.X.Y.Z` with another a link in between:

- `A.so` -> `A.so.X`;
- `A.so.X` -> `A.so.X.Y.Z`.

The `A.so` link is then part of the `devel` subpackage, which depends on the default package providing `A.X` and `A.X.Y.Z`. If the binary version doesn't have all `X.Y.Z` components (major, minor, and patch version numbers), then the middle link might fall out or the whole thing gets simplified and we're left with `A.so` -> `A.so.[some_other_versioning_scheme]`. Example in `ding-libs.spec`.

More info about conventions [here](https://medium.com/@manojddesilva/managing-linux-shared-libraries-be67ae14af2a), [here](https://unix.stackexchange.com/questions/475/how-do-so-shared-object-numbers-work), and [here](https://tldp.org/HOWTO/Program-Library-HOWTO/shared-libraries.html).